pkg/blockstorage/vmware/vmware.go

```diff
@@ -335,7 +335,7 @@ func (p *FcdProvider) SnapshotDelete(ctx context.Context, snapshot *blockstorage
 	if err != nil {
 		return errors.Wrap(err, "Cannot infer volume ID from full snapshot ID")
 	}
-	return wait.PollImmediate(time.Second, defaultRetryLimit, func() (bool, error) {
+	return wait.PollUntilContextTimeout(ctx, time.Second, defaultRetryLimit, false, func(context.Context) (bool, error) {
```

Comment:
Does the internal `func` need a named argument for context? This is done above on line 260 - it's named `innerCtx`